packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -242,6 +258,39 @@ class Auth {
         return validTimeSeconds <= tokenAgeSeconds
     }
 
+    /**
+     * WARNING: This function is relevant to be used in Hybrid deployments only.
+     * Compares the refresh_token keys for guest('cc-nx-g') and registered('cc-nx') login from the cookie received from SFRA with the copy stored in localstorage on PWA Kit
+     * to determine if the login state of the shopper on SFRA site has changed. If the keys are different we return true considering the login state did change. If the keys are same,
+     * we compare the values of the refresh_token to cover an edge case where the login state might have changed multiple times on SFRA and the eventual refresh_token key might be same
+     * as that on PWA Kit which would incorrectly show both keys to be the same even though the sessions are different.
+     * @returns {boolean} true if the keys do not match (login state changed), false otherwise.
+     */
+    private hasSFRAAuthStateChanged() {
+        const refreshTokenKey: AuthDataKeys =
+            (this.get('refresh_token_registered') && 'refresh_token_registered') ||
+            'refresh_token_guest'
+
+        const refreshTokenCopyKey =
+            (this.get('refresh_token_registered_copy') && 'refresh_token_registered_copy') ||
+            'refresh_token_guest_copy'
+
+        if (DATA_MAP[refreshTokenKey].key !== DATA_MAP[refreshTokenCopyKey].key) {
+            return true
+        }
+
+        return this.get(refreshTokenKey) !== this.get(refreshTokenCopyKey)
+    }
+
+    /**
+     * Used to validate JWT expiry and ensure auth state consistency with SFRA in a hybrid setup
+     * @param token access_token received on SLAS authentication
+     * @returns {boolean} true if JWT is valid; false otherwise
+     */
+    private isTokenValid(token: string) {
+        return !this.isTokenExpired(token) && !this.hasSFRAAuthStateChanged()
```

Comment:
```suggestion
        return !this.isTokenExpired(token) && !this.hasSFRAAuthStateChanged(isGuest)
```

Reply:
Updated code to use isGuest to determine cookie to fetch.